tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -101,21 +163,20 @@ private void Run(BsonDocument shared, BsonDocument test)
             CreateCollection();
             InsertData(shared);
 
-            using (ConfigureFailPoint(test))
+            if (CoreTestConfiguration.Cluster.Description.Type == ClusterType.Sharded)
             {
-                var eventCapturer = new EventCapturer()
-                    .Capture<CommandStartedEvent>(e => !__commandsToNotCapture.Contains(e.CommandName));
+                PrimeShardRoutersWithDistinctCommand();
+            }
+
+            var eventCapturer = new EventCapturer()
+                .Capture<CommandStartedEvent>(e => !__commandsToNotCapture.Contains(e.CommandName));
 
+            var useMultipleShardRouters = test.GetValue("useMultipleMongoses", false).AsBoolean;
```

Comment:
AsBoolean should be fine here but as a general rule we call ToBoolean(), just in case the truth value is encoded as something other than true, as in:

    useMultipleMongoses : 1

That's important when dealing with responses from the server. Probably irrelevant here since the JSON files all currently use true.